Sming/Wiring/IpAddress.h

```diff
@@ -49,73 +55,81 @@ class IpAddress : public Printable
 
 	IpAddress(uint8_t first_octet, uint8_t second_octet, uint8_t third_octet, uint8_t fourth_octet)
 	{
-		IP4_ADDR(&address, first_octet, second_octet, third_octet, fourth_octet);
+		IP_ADDR4(&address, first_octet, second_octet, third_octet, fourth_octet);
 	}
 
 	IpAddress(uint32_t address)
 	{
-		this->address.addr = address;
+		ip_addr_set_ip4_u32(&this->address, address);
 	}
 
-	IpAddress(ip_addr address)
+	IpAddress(ip_addr_t &addr)
 	{
-		this->address.addr = address.addr;
+		address = addr;
 	}
 
-#if LWIP_VERSION_MAJOR == 2
-	IpAddress(ip_addr_t address)
+	IpAddress(const ip_addr_t &addr)
+	{
+		address = addr;
+	}
+
+#if LWIP_VERSION_MAJOR == 2 && LWIP_IPV6
+	IpAddress(ip4_addr_t &addr)
 	{
-		this->address = address;
+		ip_addr_copy_from_ip4(address, addr);
 	}
 #endif
 
 	IpAddress(const uint8_t* address)
 	{
-		IP4_ADDR(&this->address, address[0], address[1], address[2], address[3]);
+		IP_ADDR4(&this->address, address[0], address[1], address[2], address[3]);
```

Comment:
Could change to `IpAddress(const uint8_t[4])` to clarify. Could probably also do with comment for constructor.
```
/*
@param address IP4 address octets
*/
```